fishtank/src/cluster.ts

```diff
@@ -27,11 +54,55 @@ export class Cluster {
     return this.backend.createNetwork(this.networkName(), { attachable: true, internal: true })
   }
 
-  async spawn(options: { name: string; image?: string }): Promise<void> {
+  async spawn(options: { name: string; image?: string; config?: NodeConfig }): Promise<void> {
     const name = this.containerName(options.name)
-    return this.backend.runDetached(options.image ?? DEFAULT_IMAGE, {
+
+    const args: string[] = []
+    const runOptions: RunOptions = {
       name,
       networks: [this.networkName()],
-    })
+    }
+
+    if (options?.config) {
```

Comment:
Note: this creates a directory that is never removed. I suggest to remove it as part of `teardown()`. (This can be done in a future PR)

Reply:
okay, will be in part 2.